hail_scripts/reference_data/combine.py

```diff
@@ -0,0 +1,123 @@
+    """
+    select_fields = {}
+    if selects is None:
+        return select_fields
+    if isinstance(selects, list):
+        select_fields = {selection: base_ht[selection] for selection in selects}
+    elif isinstance(selects, dict):
+        for key, val in selects.items():
+            # Grab the field and continually select it from the hail table.
+            ht = base_ht
+            for attr in val.split('.'):
+                # Select from multi-allelic list.
+                if attr.endswith('#'):
+                    attr = attr[:-1]
+                    ht = ht[attr][base_ht.a_index - 1]
+                else:
+                    ht = ht[attr]
+            select_fields[key] = ht
+    return select_fields
+
+def get_custom_select_fields(custom_select, ht):
+    if custom_select is None:
+        return {}
+    return custom_select(ht)
+
+def get_enum_select_fields(enum_selects, ht):
+    enum_select_fields = {}
+    if enum_selects is None:
+        return enum_select_fields
+    for field_name, values in enum_selects.items():
+        lookup = hl.dict(hl.enumerate(values, index_first=False))
+        # NB: this conditioning on type is "outside" the hail expression context.
+        if (
+            isinstance(ht[field_name].dtype, (hl.tarray, hl.tset)) and 
+            ht[field_name].dtype.element_type == hl.tstr
+        ):
+            enum_select_fields[f'{field_name}_ids'] = ht[field_name].map(lambda x: lookup[x])
+        else:
+            enum_select_fields[f'{field_name}_id'] = lookup[ht[field_name]]
+    return enum_select_fields
+
+def get_ht(dataset: str, reference_genome: str):
+    config = CONFIG[dataset][reference_genome]
+    field_name = config.get('field_name') or dataset
+    ht = hl.read_table(config['path']).distinct()
+    ht = ht.filter(config['filter'](ht)) if 'filter' in config else ht
+    ht = ht.select(**{
+        **get_select_fields(config.get('select'), ht),
+        **get_custom_select_fields(config.get('custom_select'), ht),
+    })
+    ht = ht.transmute(**get_enum_select_fields(config.get('enum_select'), ht))
+    return ht.select(**{field_name: ht.row.drop(*ht.key)})
+
+def update_joined_ht_globals(
+    joined_ht, datasets, version, coverage_datasets, reference_genome
+):
+    # Track the dataset we've added as well as the source path.
+    included_dataset = {
+        k: v[reference_genome]['path']
+        for k, v in CONFIG.items()
+        if k in datasets + coverage_datasets
+    }
+    enum_definitions = {
+        k: {enum_field_name: enum_values}
+        for k, v in CONFIG.items()
+        if k in datasets + coverage_datasets
+        if 'enum_select' in v[reference_genome]
+        for enum_field_name, enum_values in v[reference_genome]['enum_select'].items()
+    }
+    # Add metadata, but also removes previous globals.
+    return joined_ht.select_globals(
+        date=datetime.now().isoformat(),
+        datasets=hl.dict(included_dataset),
```

Comment:
Do you think it makes sense to add in functionality to get a specific datasets version? I know v02 grabs clinvar's version during the join in the loading pipeline but moving forward with one massive table, we'll want to keep version per dataset. This is probably out of scope for this but just a thought when I saw the globals. It should probably be added to this dict.

Reply:
Yep!  I think that would be a useful feature.  `v03` has that for clinvar as well.  I think all the other datasets are being treated as statically versioned though.

Reply:
I think it would be nice to have a per-dataset version! I don't want to blow up the scope of what you are working on, but that would be really helpful in the future